### PR TITLE
gstreamer1.0-plugins-good: enable rpicamsrc plugin

### PR DESCRIPTION
**Pull request by quaresmajose**
Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

**- What I did**

**- How I did it**

### CODE REVIEW

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good/0001-rpicamsrc-add-vchostif-library-as-it-is-required-to-.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+From: Jose Quaresma <quaresma.jose@gmail.com>
+Date: Sun, 15 Nov 2020 11:30:07 +0000
+Subject: [PATCH] rpicamsrc: add vchostif library as it is required to build
+ successful
+
+fix: undefined reference to `vc_gencmd'
+/usr/src/debug/gstreamer1.0-plugins-good/1.18.1-r0/build/../gst-plugins-good-1.18.1/sys/rpicamsrc/RaspiCamControl.c:1440: undefined reference to `vc_gencmd'
+
+Upstream-Status: Submitted [https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/merge_requests/818]
+
```
Comment (agherzan) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-good/0001-rpicamsrc-add-vchostif-library-as-it-is-required-to-.patch:11]: NIT: remove newline in between footers.

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.%.bbappend [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FILESEXTRAPATHS_prepend_rpi := "${THISDIR}/gstreamer1.0-plugins-good:"
```
Comment (agherzan) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.%.bbappend:1]: NIT: Take advantage of PN.

**Review by agherzan — APPROVED** [review]
Looks good. Thanks. Would it make sense to squash the commits?

**Comment by agherzan** [comment]
@agherzan-ci ok to test

**Review by quaresmajose — COMMENTED** [review]
> Looks good. Thanks. Would it make sense to squash the commits?

I have two commits because I have submitted the first patch to OE-core and this can be reverted if it is accepted.

I still have a doubt about the second patch.
Is the `userland` package incompatible for some reason when the `vc4graphics` is defined on MACHINE_FEATURES ?

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.%.bbappend [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+SRC_URI_append_rpi = " file://0001-rpicamsrc-add-vchostif-library-as-it-is-required-to-.patch"
+
+EXTRA_OEMESON_remove_rpi = "-Drpicamsrc=disabled"
+PACKAGECONFIG[rpi] = "-Drpicamsrc=enabled,-Drpicamsrc=disabled,userland"
+
+PACKAGECONFIG_append_rpi = " ${@bb.utils.contains('MACHINE_FEATURES', 'vc4graphics', '', 'rpi', d)}"
```
[the remark this answers is not in the document]
Comment (agherzan) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.%.bbappend:8]: Yes. When vc4 is selected, mesa is used for the gl libraries.

https://github.com/agherzan/meta-raspberrypi/blob/master/conf/machine/include/rpi-default-providers.inc

[the remark this answers is not in the document]
Comment (kraj) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.%.bbappend:8]: add a space before the first quote then you only added space when needed unlike adding it always something like
```
PACKAGECONFIG_append_rpi = "${@bb.utils.contains('MACHINE_FEATURES', 'vc4graphics', '', ' rpi', d)}"
```

[the remark this answers is not in the document]
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.%.bbappend]: good point! thanks

**Review by agherzan — APPROVED** [review]
Looks good. Merged.

**Comment by shr-project** [comment]
The patch was merged in oe-core, time to revert the 1st commit. Will send PR with that now.

**Pull request merged**